freud/density.pyx

```diff
@@ -108,10 +108,26 @@ cdef class FloatCF(Compute):
                 NeighborList to use to find bonds (Default value =
                 :code:`None`).
         """
+        exclude_ii = points is None
+
+        if (points is None) != (values is None):
+            raise ValueError("Either both points and values have to be"
+                             "provided or both points and values have to"
+                             "be None")
```

Comment:
Actually, I think users should be able to pass in values as a kwarg to take advantage of automatic `exclude_ii`; the check should only be enforcing that they don't pass in a second set of points without values, which I think is more nonsensical.